### DISCUSSION TITLE
Exposure time with CameraCapture

### DISCUSSION THREAD

**Discussion by PiezaKake** [bonsai-rx/bonsai#1334] [Q&A]
I am using CameraCapture to record at 250 fps. The infrared light I am using is causing flickering in the recorded video. When I record with a different software and set the exposure time at 1/306 sec, the flickering is gone and the camera records at 250 fps. With CameraCapture, I am not sure how to properly adjust the exposure time. With the fps at 250, if I set the Exposure at -7 the flickering is gone, but the recorded frames drop to about half. If I set the exposure to -8, I record at 250 fps but with flickering. Any advice as to how to record at 250 fps without flickering?

**Comment by bruno-f-cruz** [comment]
the best way to get rid of flickering is to either strobe the lights at the frequency of the camera or acquire the frames locked to the flicker frequency of the lights. Does your camera support some sort of GPIO capabilities?

Aside from that, if you make the exposure long enough you might be able to average over the flicker, but I don't think reducing the exposure will help you that much. 

Regardless, this documentation might help with setting the values:

https://www.kurokesu.com/main/2020/05/22/uvc-camera-exposure-timing-in-opencv/
